### PR TITLE
Fix DateTime JsonConverter and add test

### PR DESCRIPTION
**Pull request by crobibero**

### CODE REVIEW

**Review by thornbill — APPROVED** [review]

File: MediaBrowser.Common/Json/Converters/JsonDateTimeIso8601Converter.cs [hunk 1]
```diff
@@ -19,6 +19,10 @@ public override DateTime Read(ref Utf8JsonReader reader, Type typeToConvert, Jso
 
         /// <inheritdoc />
         public override void Write(Utf8JsonWriter writer, DateTime value, JsonSerializerOptions options)
-            => writer.WriteStringValue(value.ToString("O", CultureInfo.InvariantCulture));
+            // Must specify DateTime.Kind so the DateTime is properly formatted.
+            // Expected output 2000-01-01T00:00:00.0000000Z
+            => writer.WriteStringValue(
+                DateTime.SpecifyKind(value, DateTimeKind.Utc)
```
Comment (cvium) [inline MediaBrowser.Common/Json/Converters/JsonDateTimeIso8601Converter.cs:25]: What if a datetime is not actually UTC? 🤔

File: tests/Jellyfin.Common.Tests/Json/JsonIso8601ConverterTests.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+using System.Text.Json;
+using MediaBrowser.Common.Json.Converters;
+using Xunit;
+
+namespace Jellyfin.Common.Tests.Json
+{
+    public static class JsonIso8601ConverterTests
+    {
+        [Fact]
+        public static void Serialize_Success()
```
Comment (cvium) [inline tests/Jellyfin.Common.Tests/Json/JsonIso8601ConverterTests.cs:11]: Bad test name. Not descriptive enough.

**Review by cvium — CHANGES_REQUESTED** [review]
See code comments...

File: MediaBrowser.Common/Json/Converters/JsonDateTimeIso8601Converter.cs (hunk 1 above)
Comment (cvium) [reply MediaBrowser.Common/Json/Converters/JsonDateTimeIso8601Converter.cs:25]: This looks like a non-issue.

**Review by BaronGreenback — APPROVED** [review]

**Pull request closed without merging**